waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
+                "vehicle_SE3_camera_": {
+                    "rotation": {"coefficients": [qw, qx, qy, qz]},
+                    "translation": [x, y, z],
+                },
+            },
+        }
+        calib_dict["camera_data_"] += [cam_dict]
+    return calib_dict
+
+
+def dump_pose(
+    city_SE3_egovehicle: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves the SE3 transformation from city frame to egovehicle frame at a particular timestamp
+
+    Args:
+        city_SE3_egovehicle: A (4,4) numpy array representing the SE3 transformation from city to egovehicle frame
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    x, y, z = city_SE3_egovehicle[:3, 3]
+    R = city_SE3_egovehicle[:3, :3]
+    assert np.allclose(city_SE3_egovehicle[3], np.array([0, 0, 0, 1]))
+    q = rotmat2quat(R)
+    w, x, y, z = q
+    pose_dict = {"rotation": [w, x, y, z], "translation": [x, y, z]}
+    json_fpath = f"{parent_path}/{log_id}/poses/city_SE3_egovehicle_{timestamp}.json"
+    check_mkdir(str(Path(json_fpath).parent))
+    save_json_dict(json_fpath, pose_dict)
+
+
+def dump_point_cloud(
+    points: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves point cloud as .ply file extracted from Waymo's range images
+
+    Args:
+        points: A (N,3) numpy array representing the point cloud created from lidar readings
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    data = {"x": points[:, 0], "y": points[:, 1], "z": points[:, 2]}
+    cloud = PyntCloud(pd.DataFrame(data))
+    cloud_fpath = f"{parent_path}/{log_id}/lidar/PC_{timestamp}.ply"
+    check_mkdir(str(Path(cloud_fpath).parent))
+    cloud.to_file(cloud_fpath)
+
+
+def dump_object_labels(
+    labels: List[waymo_open_dataset.label_pb2.Label],
+    timestamp: int,
+    log_id: str,
+    parent_path: str,
+    track_id_dict: Dict,
+) -> None:
+    """Saves object labels from Waymo dataset as json files
+
+    Args:
+        labels: A list of Waymo labels
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+        track_id_dict: Dictionary to store object ID to track ID mappings
+    """
+    argoverse_labels = []
+    for label in labels:
+        argoverse_labels.append(build_argo_label(label, timestamp, track_id_dict))
+    json_fpath = f"{parent_path}/{log_id}/per_sweep_annotations_amodal/tracked_object_labels_{timestamp}.json"
+    check_mkdir(str(Path(json_fpath).parent))
+    save_json_dict(json_fpath, argoverse_labels)
+
+
+def build_argo_label(
```

Comment:
can we add a simple unit test/smokescreen on this function, just to make sure it runs without crashing?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/9db2006d2741dbd66d59c22e07604a9bd62fa2a6)